src/Azure.Functions.Cli/Actions/KubernetesActions/FunctionMetadataGenerator.cs

```
@@ -0,0 +1,174 @@
+                    bindings.Add(new Dictionary<string, object>
+                {
+                    {"type", "queueTrigger"},
+                    {"direction", "In"},
+                    {"name", properties.GetValueOrDefault("arg_name", "message")},
+                    {"queueName", properties["queue_name"]},
+                    {"connection", properties["connection"]},
+                    {"properties", new Dictionary<string, object> { { "supportsDeferredBinding", "True" } } }
+                });
+                    break;
+                case "service_bus_queue_trigger":
+                    bindings.Add(new Dictionary<string, object>
+                {
+                    {"type", "serviceBusTrigger"},
+                    {"direction", "In"},
+                    {"name", properties.GetValueOrDefault("arg_name", "message")},
+                    {"queueName", properties["queue_name"]},
+                    {"connection", properties["connection"]},
+                    {"cardinality", "One"},
+                    {"properties", new Dictionary<string, object> { { "supportsDeferredBinding", "True" } } }
+                });
+                    break;
+                case "service_bus_topic_trigger":
+                    bindings.Add(new Dictionary<string, object>
+                {
+                    {"type", "serviceBusTrigger"},
+                    {"direction", "In"},
+                    {"name", properties.GetValueOrDefault("arg_name", "message")},
+                    {"topicName", properties["topic_name"]},
+                    {"subscriptionName", properties["subscription_name"]},
+                    {"connection", properties["connection"]},
+                    {"cardinality", "One"},
+                    {"properties", new Dictionary<string, object> { { "supportsDeferredBinding", "True" } } }
+                });
+                    break;
+                    // Add additional cases for other trigger types if necessary
+            }
```

If a new or unknown trigger type is encountered, it silently does nothing. This can cause hard-to-debug issues. IMHO extra logging will help here.